src/main/scala/com/atomist/tree/content/text/OverwritableTextTreeNode.scala

```
@@ -0,0 +1,241 @@
+    invalidator = by
+    invalidateChildren(by)
+    state = Invalidated
+  }
+
+  private var invalidator: OverwriteableTextTreeNodeParent = _
+
+  @ExportFunction(readOnly = false, description = "Update the whole value")
+  override def update(to: String): Unit = requireReady {
+    _value = to
+    invalidateChildren(this)
+    _parent.commit()
+    state = Overridden
+  }
+
+  /*
+   * Normal TreeNode methods
+   */
+  override val nodeTags: Set[String] = additionalTags + TreeNode.Dynamic + "OverwriteableText"
+
+  override val nodeName: String = name
+
+  override val significance = Signal
+
+  override def value: String = requireNotInvalidated(_value)
+
+  override def childNodeNames: Set[String] = requireNotInvalidated(requireNotOverridden(visibleChildren.map(_.nodeName).toSet))
+
+  override def childNodeTypes: Set[String] = requireNotInvalidated(requireNotOverridden(visibleChildren.flatMap(_.nodeTags).toSet))
+
+  override def childrenNamed(key: String): Seq[TreeNode] = requireNotInvalidated(requireNotOverridden(visibleChildren.filter(_.nodeName == key)))
+
+  override def childNodes: Seq[TreeNode] = requireNotInvalidated(requireNotOverridden(visibleChildren))
+
+  override def address: String = requireParentalKnowledge(_parent.address + "/" + locationStep)
+
+  override def commit(): Unit = requireReady {
+    _value = allKids.map(_.value).mkString("") // some child has changed
+    _parent.commit()
+  }
+
+  @ExportFunction(readOnly = true, description = "Return the parent node")
+  def parent: TreeNode = {
+    _parent match {
+      case o: OverwriteableTextInFile => o.fileView
+      case _ => _parent
+    }
+  }
+
+  @ExportFunction(readOnly = true, description = "Return the format info for the start of this structure in the file or null if not available")
+  final def formatInfo: FormatInfo =
+    rootNode.formatInfo(this).orNull // null is only error cases right? why not throw?
```

Yes now it would be better to throw an exception